signbank/dictionary/update.py

```diff
@@ -764,6 +774,316 @@ def confirm_import_gloss_csv(request):
+            if 'dataset_id' in request.session: del request.session['dataset_id']
+            if 'glosses_new' in request.session: del request.session['glosses_new']
+            # Set a message to be shown so that the user knows what is going on.
+            messages.add_message(request, messages.WARNING, _('Cancelled adding CSV data.'))
+            return HttpResponseRedirect(reverse('dictionary:import_nzsl_share_gloss_csv'))
+
+        elif 'confirm' in request.POST:
+            glosses_added = []
+            dataset = None
+            translations = []
+            comments = []
+            videos = []
+            new_glosses = {}
+            bulk_create_gloss = []
+            bulk_update_glosses = []
+            bulk_semantic_fields = []
+
+            if 'glosses_new' and 'dataset_id' in request.session:
+                dataset = Dataset.objects.get(id=request.session['dataset_id'])
+                language_en = Language.objects.get(name="English")
+                language_mi = Language.objects.get(name="Māori")
+                gloss_content_type = ContentType.objects.get_for_model(Gloss)
+                video_type = FieldChoice.objects.get(field="video_type", english_name="validation")
+                site = Site.objects.get_current()
+                comment_submit_date = datetime.datetime.now(tz=get_current_timezone())
+                semantic_fields = FieldChoice.objects.filter(
+                    field="semantic_field"
+                ).values_list("english_name", "pk")
+                semantic_fields_dict = {field[0]: field[1] for field in semantic_fields}
+
+                for row_num, gloss in enumerate(request.session['glosses_new']):
+                    # will iterate over these glosses again after bulk creating and to make sure
+                    # we preserve order
+                    new_glosses[str(row_num)] = gloss
+                    bulk_create_gloss.append(Gloss(
+                        dataset=dataset,
+                        # need to make idgloss unique in dataset,
+                        # but gloss word can appear in multiple rows, so
+                        # idgloss will be updated to word:pk in second step
+                        idgloss=f"{gloss['word']}_row{row_num}",
+                        idgloss_mi=gloss.get("maori", None),
+                        notes=gloss.get("notes", ""),
+                        created_by=request.user,
+                        updated_by=request.user
+                    ))
+
+                bulk_created = Gloss.objects.bulk_create(bulk_create_gloss)
+                idglosses = [x.idgloss for x in bulk_created]
+                bulk_created_with_pk = Gloss.objects.filter(
+                    idgloss__in=idglosses
+                )
+
+                gloss_pks = []
+
+                for gloss in bulk_created_with_pk:
+                    gloss_pks.append(gloss.pk)
+                    old_id_gloss = gloss.idgloss.split("_row")
```

Comment:
**suggestion**: since it's assumed to be a 2-tuple, you could label the parts by assigning them to variables here instead of using the indices:

```
raw_idgloss, row_number = ...
```